webview.h

```diff
@@ -108,7 +108,7 @@ WEBVIEW_API void webview_bind(webview_t w, const char *name,
                               void *arg);
 
 // Removes a native C callback that was previously set by webview_bind.
-WEBVIEW_API void webview_unbind(webview_t w, const char *name);
+WEBVIEW_API void *webview_unbind(webview_t w, const char *name);
```

Comment:
Correct... Here, the user can expect a return if they need to free a pointer, but they will get no errors if the binding was not found - just nullptr. Unfortunately, I'm not quite sure of the best way to fix this.

Reply:
@justjosias suggested adding a new function to the C API to retrieve the `arg` in #730. Current users of the library will have to modify their code anyway to get the `arg` so I do not see it as a problem. If there is a real use case for retrieving the `arg` then to me it makes sense to split that feature into a dedicated function. That way both `webview_unbind` and something like `webview_get_bound_arg` can return an error code.

Pseudocode:

```c
typedef struct {} user_arg_t;
user_arg_t *user1, *user2;
webview_error_t err;

err = webview_bind(w, "log", cb, user1);
err = webview_get_bound_arg(w, "log", &user2);
err = webview_unbind(w, "log");
assert(user1 == user2);
```

Reply:
If webview_get_bound_arg uses pass-by-reference and returns an error code in your example, couldn't we just do that with unbind? The only time a user needs access to the arg is when unbinding, so the extra function seems kind of like it could be a hassle.

Reply:
Are you thinking something like this?

```c
typedef struct { ... } user_arg_t;
void log(...) {}
user_arg_t *user1, *user2;
webview_error_t err;

err = webview_bind(w, "log", log, user1);
err = webview_unbind(w, "log", &user2);
assert(user1 == user2);
```

Reply:
Yes, exactly! How do you feel about it becoming:
`webview_error_t webview_unbind(webview_t w, const char *name, void *);`
Then we can consistently return error codes.

Reply:
We might be able to retain ABI compatibility with version checks in the library and only return the associated arg value (via output parameter) if the user's header is at least as recent as the implementation. If we go this route then may I suggest that we make the change painless for existing users?

Reply:
I think for this change, since we are mitigating a memory leak, we might not want it to be ABI compatible. That way, users have to recompile if they want the latest version which will encourage them to fix the current memory leak situation. Does the version check imply that we need to bump the library version for this change?
Maybe we should worry about ABI compatibility after we implement the error codes in #766, because that will move webview toward a more standard API.

Reply:
Unbinding is a relatively new feature. I'd be fine with simply changing the API here.

Reply:
@dandeto Just so that I am crystal clear on this: The mentioned leaks are not caused by the library itself, but instead those callers who aren't properly managing the memory they themselves allocated and passed into `webview_bind` by letting `webview_unbind` return the same `arg` back to the caller. Is that correct?

@justjosias Unbinding has been in the code base since February 20, 2022 (more than 9 months). Is that OK?

Reply:
@dandeto

> Does the version check imply that we need to bump the library version for this change?

Given the amount of time without a single release since the library rewrite, one could consider releasing a compatible version 0.10 officially before breaking anything even though it doesn't tick all the boxes. Then one could be more flexible with the following 0.x releases and even break things with good reasons.

Reply:
> Unbinding has been in the code base since February 20, 2022 (more than 9 months). Is that OK?

I believe it would be, considering it has been undocumented and we were open to changing it afterward before development was suspended again. I don't know of any bindings that implement it yet. I'm willing to hear other ideas though, especially if third parties are already using it.

Reply:
I took a brief look at the bindings listed in the readme and found that the following 3 uses `webview_unbind`:

* https://github.com/malisipi/mui
* https://github.com/Winterreisender/webviewko
* https://github.com/webview/webview_deno